.changeset/late-bulldogs-help.md

```diff
@@ -0,0 +1,7 @@
+---
+"wrangler": patch
+---
+
+feat: Add support for the `nodejs_compat` Compatibility Flag when bundling a Worker with Wrangler
+
+This new Compatibility Flag is incompatible with the legacy `--node-compat` CLI argument and `node_compat` configuration option. If you want to use the new runtime Node.js compatibility features, please remove the `--node-compat` argument from your CLI command or your config file.
```

Comment:
I'm sure we've thought about this a bit already, so don't want to make others rehash past discussion, but I'm curious what we're trying to guard against?

In my mind, if both flags are enabled, then that should mean:

1. Node imports that are built into workerd aren't polyfilled (new `nodejs_compat`)
2. Node imports that aren't built into workerd are polyfilled (old `node_compat`)

...such that people who today need a mix of node APIs that are / aren't in workerd can start using our built-in support right now, and don't have to wait until all the node APIs they need are available in workerd, to make the switch.

Reply:
The polyfills are just generally sub-par. We should deprecate ASAP imo, and this is a totally fair forcing function. Just taking advantage of it to encourage people to move over to the proper `nodejs_compat` compatibility flag.